libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -726,16 +765,35 @@ string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type)
 			function <functionName>(array, value) {
 				let oldLen := <fetchLength>(array)
 				if iszero(lt(oldLen, <maxArrayLength>)) { invalid() }
-				sstore(array, add(oldLen, 1))
+				<?isByteArray>
+					let data := sload(array)
+					switch oldLen
+					case 31 {
+						// Here we have special case when array switches from short array to long array
+						// We need to copy data
+						let dataArea := <dataAreaFunction>(array)
+						sstore(dataArea, <shr>(data))
```

Comment:
Are you sure this is correct?

Reply:
Rightmost byte is length, so I am shifting to right here, so I clear that byte. Than leftmost (which is 31st offset item in slot) will be left open for the push we are doing right now. Also, I think we should make this to work the same way like old codegen, so the leftmost is offset 0, and rightmost offset item 31. Also, that way we will not need to shift here.

Reply:
Ah ok! I thought there was something weird here! Yes, it should be exactly as in the old codegen.

Reply:
Ok, I'll do that change in this PR than